mkosi

```diff
@@ -564,6 +564,18 @@ def mount_image(args, workspace, loopdev, root_dev, home_dev, srv_dev, root_read
 
             umount(root)
 
+@complete_step("Assigning hostname")
+def assign_hostname(args, workspace):
+    root = os.path.join(workspace, "root")
+    hostname_path = os.path.join(root, "etc/hostname")
+    
+    if not args.hostname:
+        if os.path.isfile(hostname_path):
+            os.remove(hostname_path)
+    else:
+        with open(hostname_path, "w+") as f:
```

Comment:
@dteleguin didn't make attention to this one yesterday :) Yes, that's not right

I think @poettering just was doing some different business by the time of comment :) But anyway I agree to remove /etc/hostname in any cases (when nothing is given and before assignment).